src/vm.rs

```diff
@@ -90,6 +91,42 @@ pub(crate) enum VmExecutionResult {
     },
 }
 
+// https://docs.soliditylang.org/en/v0.8.26/internals/layout_in_storage.html#mappings-and-dynamic-arrays
+fn get_erc20_balance_slot(address: Address) -> U256 {
+    let mut buf = [0u8; 64];
+    buf[12..32].copy_from_slice(address.as_slice());
+    keccak256(buf).into()
+}
+
+#[derive(Debug)]
+enum LazyStrategy {
+    None,
+    RawTransfer,
+    ERC20Transfer {
+        sender_balance_slot: U256,
+        recipient_balance_slot: U256,
+    },
+}
+
+impl LazyStrategy {
+    fn from(tx_sender: &Address, tx_recipient_code_hash: &Option<B256>, input: &Bytes) -> Self {
+        if tx_recipient_code_hash.is_none() {
+            return LazyStrategy::RawTransfer;
+        };
+
+        // TODO: We cannot blindly trust method_id. Use a whitelist of tx_recipient_code_hash.
```

Comment:
Yeah, this should be configurable by users. Some use cases (that don't take external transactions) can trust all, some can whitelist, some can just turn off lazy ERC-20.

Reply:
I added `PevmChain::is_erc20_transfer`:

```
/// Check if a tx is ERC20 transfer
fn is_erc20_transfer(&self, tx: &TxEnv) -> bool;
```

If a user wants to customize the logic, they can construct a `PevmChain` by themself.